### DISCUSSION TITLE
Multi-window Windows application - How do I find out which window i am currently on?

### DISCUSSION THREAD

**Discussion by lvendramini** [dotnet/maui#14547] [Q&A]
Hi,

I am trying to see which window of a multi-window application i am on.  Microsoft documentation 
https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/windows?view=net-maui-7.0#multi-window-support 
gives the example

// Close a specific window
Application.Current.CloseWindow(secondWindow);

// Close the active window
Application.Current.CloseWindow(GetParentWindow());

But i can't get GetParentWindow() to work.   Is there a way to see which window the user is interacting with?

**Comment by keyung** [comment]
//Get default windows
 var window = Application.Current.Windows.LastOrDefault();

 App.Current.CloseWindow(window);

**Reply by holecekp** [reply]
Thank you. That helped me a lot at least for the most common scenarios. Hower, the last created Window isn't probably always the same as the active Window when the users switches among the app instances in the task manager. Does anyone know how to find the currently opened Window for Android from MainActivity?

I need to access Shell.Current from the MainActivity. The problem is that any access to Shell.Current ends up in an exception if there are multiple instances of the app opened because Shell.Current requires the developers to find out the active Window them self and access the Shell from it in that case.
